### PR TITLE
Simple tests for language server's parser

### PR DESCRIPTION
**Pull request by Nykseli**
## Detailed description

Having a test suite for the language server's parser is an easy way to catch regressions.
This testing method is currently really naive and lacks a lot of test cases, but it's a good base for us to build a good and robust test suite.

Please note that test file containing valid syntax that are prefixed with `fixme_` are currently ignored. So CI won't complain on every PR.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/mangrove-lang/mangrove-vscode/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/mangrove-lang/mangrove-vscode/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds and lints cleanly
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do
* [x] I've made sure that my commits are [signed/verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +48,6 @@
 		"vscode-languageserver-textdocument": "^1.0.7"
 	},
 	"devDependencies": {
-		"@types/jest": "^28.1.3",
```
Comment (dragonmux) [inline package.json:51]: Is there a particular reason this needed removing? Thought it'd be necessary so we get type help from the VSCode TS lang server

File: test/server/parser/parser.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+
```
Comment (dragonmux) [inline test/server/parser/parser.ts:1]: Did you mean to miss the license and authorship information from the top of this file?

**Review by dragonmux — CHANGES_REQUESTED** [review]
Generally looks good! The only two things we spotted in review are very minor.

We know the set of test cases is very much non-exhaustive, but as you went through several of the function-y things, one set of cases stuck out like a sore thumb - testing that two functions declared in the same scope and named the same thing but with parameters of different types or differing return types parse w/o error.

We should probably also test for cases like naming a function and a class the same thing should fail parsing (though syntactically fine)

File: package.json (hunk 1 above)
Comment (Nykseli) [reply package.json:51]: `@types/jest` only adds jest functions to global namespace and it has clashing type definitions with `@jest/globals`, so it's easier to just remove `@types/jest` and import the needed functions from `@jest/globals`.

The issue talking about the issue seems to be resolved but I didn't have any luck updating the dependencies.
https://github.com/facebook/jest/issues/13199

File: test/server/parser/parser.ts (hunk 2 above)
Comment (Nykseli) [reply test/server/parser/parser.ts:1]: That was a mistake on my part. Updating the PR!

**Comment by Nykseli** [comment]
All review items should now be addressed.

There's plenty of more tests that need to be done.
My goal is to improve the test suite a bit more so contributors can easily add more test cases as they come up with them!

File: package.json (hunk 1 above)
Comment (dragonmux) [reply package.json:51]: That's a good reason - thanks for the explanation :)

**Review by dragonmux — APPROVED** [review]
LGTM, and a massive improvement from no testing at all of the parser - should be useful for building test cases for the bootstrap compilers too.

Merging

**Pull request merged**